File: java/src/main/java/deltix/luminary/LiteralTime.java
package deltix.luminary;

public class LiteralTime extends Literal {
    private String value;

    public LiteralTime(String value) {
        super(LiteralKind.TIME);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "\"" + value + "\"";
    }

    public static final String MIN_VALUE_AS_STRING = "#MIN";
    public static final String MAX_VALUE_AS_STRING = "#MAX";
}

File: csharp/main/LiteralTime.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralTime : Literal, ILiteral<String>
	{
		public const String MinValueAsString = "#MIN";
		public const String MaxValueAsString = "#MAX";

		public LiteralTime(String value)
			: base(LiteralKind.Time)
		{
			Value = value;
		}

		public String Value { get; }

		public override String ToString()
		{
			return $"\"{Value}\"";
		}
	}
}